java/src/com/mongodb/csfle/CSFLEHelpers.java

```diff
@@ -0,0 +1,211 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.csfle;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.bson.BsonBinary;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import com.mongodb.AutoEncryptionSettings;
+import com.mongodb.ClientEncryptionSettings;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.vault.DataKeyOptions;
+import com.mongodb.client.vault.ClientEncryption;
+import com.mongodb.client.vault.ClientEncryptions;
+
+/*
+ * Helper methods and sample data for this companion project.
+ */
+public class CSFLEHelpers {
+
+    // Sample data
+    public static String SAMPLE_NAME_VALUE = "John Doe";
+    public static Integer SAMPLE_SSN_VALUE = 145014000;
+
+    public static Document SAMPLE_DOC = new Document()
+            .append("name", SAMPLE_NAME_VALUE)
+            .append("ssn", SAMPLE_SSN_VALUE)
+            .append("bloodType", "AB-")
+            .append("medicalRecords", Arrays.asList(
+                    new Document().append("weight", "180"),
+                    new Document().append("bloodPressure", "120/80")))
+            .append("insurance", new Document()
+                    .append("policyNumber", 123142)
+                    .append("provider", "MaestCare"));
+
+    // Reads the 96-byte local master key
+    public static byte[] readMasterKey(String filePath) throws Exception {
+        int numBytes = 96;
+        byte[] fileBytes = new byte[numBytes];
+
+        try (FileInputStream fis = new FileInputStream(filePath)) {
+            if (fis.read(fileBytes) < numBytes)
+                throw new Exception("Expected to read 96 bytes from file");
+        }
+        return fileBytes;
+    }
+
+    // JSON Schema helpers
+    private static String DETERMINISTIC_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic";
+    private static String RANDOM_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Random";
+
+    private static Document buildEncryptedField(String keyId, String bsonType, Boolean isDeterministic) {
+        return new Document().
+                append("encrypt", new Document()
+                        .append("bsonType", bsonType)
+                        .append("algorithm",
+                                (isDeterministic) ? DETERMINISTIC_ENCRYPTION_TYPE : RANDOM_ENCRYPTION_TYPE));
+    }
+
+    private static Document createEncryptMetadataSchema(String keyId) {
+        List<Document> keyIds = new ArrayList<Document>();
```

Comment:
IntelliJ gives all sorts of warning about explicit generic arguments.  Consider replacing them all with implicit generic arguments, e.g.

      List<Document> keyIds = new ArrayList<>();

Reply:
Thanks, will update.